src/SegmentationCameraSensor.cc

```diff
@@ -61,30 +57,39 @@ class ignition::sensors::SegmentationCameraSensorPrivate
   /// \brief Node to create publisher
   public: transport::Node node;
 
-  /// \brief Publisher to publish segmentation image
-  public: transport::Node::Publisher publisher;
+  /// \brief Publisher to publish segmentation colored image
+  public: transport::Node::Publisher coloredMapPublisher;
+
+  /// \brief Publisher to publish segmentation labels image
+  public: transport::Node::Publisher labelsMapPublisher;
 
   /// \brief Segmentation Image Msg
-  public: msgs::Image segmentationMsg;
+  public: msgs::Image coloredMapMsg;
 
-  /// \brief Topic to publish the segmentation image
-  public: std::string topicSegmentation = "";
+  /// \brief Segmentation Image Msg
+  public: msgs::Image labelsMapMsg;
 
-  /// \brief Buffer contains the segmentation map data
-  public: uint8_t *segmentationBuffer {nullptr};
+  /// \brief Topic to publish the segmentation colored map
+  public: std::string topicColoredMap = "/colored_map";
 
-  /// \brief Segmentation type (Semantic / Instance)
-  public: SegmentationType type {SegmentationType::Semantic};
+ /// \brief Topic to publish the segmentation labels map
+  public: std::string topicLabelsMap = "/labels_map";
 
-  /// \brief True if camera generates a colored map image
-  /// False if camera generates labels ids image
-  public: bool isColoredMap {false};
+  /// \brief Buffer contains the segmentation colored map data
+  public: uint8_t *segmentationColoredBuffer {nullptr};
+
+  /// \brief Buffer contains the segmentation labels map data
+  public: uint8_t *segmentationLabelsBuffer {nullptr};
```

Comment:
mmm, I can't do that, because i copy the data from the buffer coming from the rendering camera (which is in the format of raw pointer) and copy its content with memcpy (and it refuses to do that with smart pointers)